ivadomed/loader/bids_dataframe.py

```diff
@@ -119,8 +119,8 @@ def create_bids_dataframe(self):
             # and with derivative files of chosen target_suffix
             df_next = df_next[(~df_next['path'].str.contains('derivatives')
                                & df_next['suffix'].str.contains('|'.join(self.contrast_lst)))
-                              | (df_next['path'].str.contains('derivatives')
-                                 & df_next['filename'].str.contains('|'.join(self.target_suffix)))]
+                               | (df_next['path'].str.contains('derivatives')
+                               & (df_next['filename'].str.split('.').apply(lambda x: x[0])).str.endswith(tuple(self.target_suffix)))]
```

Comment:
I have to say, this is some of the most complex one liners I have seen in Python ever among all my projects. I sort of understand we have to write it mainly due to the df[ ] filtering syntax but I keep wondering if we can make it more readable by either breaking it down or simplifying the overall expression. 

My understanding is for line 123, you are taking file name, convert to string, split it by `.` into list then for the FIRST part, (pre extension?), filter for elements in that above list which ends with one of the target_suffix and applying this boolean selection to filter the df_next along with previous expression before |...

... and I only explained line 123, not 120, 121 and 122.

I think it is fine for now if that is your goal to do what I described above?  

I will ask @konantian to see if we can find a way to reword this to be more readable. I strongly recommend adding more comment explaining the mechanism and maybe best give a few example file names. 

I do agree this will be better than the earlier contain approach is MORE specific than before though. 

I will update to approval once the unit test passes.

Reply:
> I have to say, this is some of the most complex one liners I have seen in Python ever among all my projects. I sort of understand we have to write it mainly due to the df[ ] filtering syntax but I keep wondering if we can make it more readable by either breaking it down or simplifying the overall expression.
> 
> My understanding is for line 123, you are taking file name, convert to string, split it by `.` into list then for the FIRST part, (pre extension?), filter for elements in that above list which ends with one of the target_suffix and applying this boolean selection to filter the df_next along with previous expression before |...
> I think it is fine for now if that is your goal to do what I described above?

Yes, you explanation is right!

> I will ask @konantian to see if we can find a way to reword this to be more readable. I strongly recommend adding more comment explaining the mechanism and maybe best give a few example file names.

I agree, this one liner is hard to understand, any help to simplify it would by greatly appreciated!
In the meantime, I updated the comment above the command with more details on the behavior and a filename example for each, let me know if that is better (or worse 😅)